backend/common_python/common_model/l4ma.py

```
@@ -611,6 +262,25 @@ def __init__(self, config: L4maArch, backend: L4maBackend):
         self.config = config
         self.backend = backend
 
+        # Force deterministic behavior to debug MPS non-determinism
+        import os
+        if os.environ.get('PIE_METAL_DEBUG_HIDDEN_STATES') == '1':
+            torch.manual_seed(42)
+            if hasattr(torch, 'use_deterministic_algorithms'):
+                try:
+                    torch.use_deterministic_algorithms(True, warn_only=True)
+                    print("[DEBUG L4maModel] Enabled deterministic algorithms (warn_only=True)")
+                except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)